File: ecommerce-frontend/src/api/productService.js
import api from "./api";

export const getAllProducts=()=>{
    return api.get("/products");
}

export const addProduct = (productData) => {
    return api.post("/products", productData);
};

export const getProductById = (id) => {
    return api.get(`/products/${id}`);
};

export const updateProduct = (id, product) => {
    return api.put(`/products/${id}`, product);
};

export const deleteProduct = (id) => {
    return api.delete(`/products/${id}`);
};

export const getProductsByCategory = (category) => {

    return api.get(`/products/category/${category}`);

};

export const searchProducts = (keyword) => {

    return api.get(

        `/products/search?keyword=${keyword}`

    );

};

File: ecommerce-site-backend/src/main/java/com/raja/ecommerce_site/service/ProductService.java
package com.raja.ecommerce_site.service;

import com.raja.ecommerce_site.dto.ProductRequest;
import com.raja.ecommerce_site.dto.ProductResponse;

import java.util.List;

public interface ProductService {

    ProductResponse addProduct(ProductRequest request);

    List<ProductResponse> getAllProducts();

    ProductResponse getProductById(Long id);

    ProductResponse updateProduct(Long id, ProductRequest request);

    void deleteProduct(Long id);

    List<ProductResponse> getProductsByCategory(String category);

    List<ProductResponse> searchProducts(String keyword);
}